[BUG] Unconditional definition of __APPLE_USE_RFC_3542 on macOS

**Describe the bug**
On macOS, to get IPv6 compliance on some otherwise unsupported socket options, we need to define `__APPLE_USE_RFC_3542` before including system networking headers.

This is done in `srt/platform_sys.h`. However, this is done unconditionally. It the application is doing some IPv6 somewhere else and already defined it, we get a "macro redefined" error:

~~~
In file included from /opt/homebrew/include/srt/srt.h:37:
/opt/homebrew/include/srt/platform_sys.h:71:9: error: '__APPLE_USE_RFC_3542' macro redefined [-Werror,-Wmacro-redefined]
   71 | #define __APPLE_USE_RFC_3542 /* IPV6_PKTINFO */
      |         ^
~~~

Please enclose this into something like:
~~~
#if !defined(__APPLE_USE_RFC_3542)
    #define __APPLE_USE_RFC_3542 /* IPV6_PKTINFO */
#endif
~~~

**Desktop (please provide the following information):**
 - OS: macOS 15.1.1
 - SRT Version / commit ID: 1.5.4

--- a/srtcore/platform_sys.h
+++ b/srtcore/platform_sys.h
@@ -68,7 +68,9 @@
 // also other macros, like TARGET_OS_IOS etc.
 
 #include "TargetConditionals.h"
+#ifndef __APPLE_USE_RFC_3542
 #define __APPLE_USE_RFC_3542 /* IPV6_PKTINFO */
+#endif
 
 #ifdef SRT_IMPORT_TIME
       #include <mach/mach_time.h>

